File: Serene/Serene.Script/Administration/Role/RoleGrid.cs
using jQueryApi;
using Serenity;
using System.Runtime.CompilerServices;

namespace Serene.Administration
{
    [Imported]
    public class RoleGrid : EntityGrid<RoleRow>
    {
        public RoleGrid(jQueryObject container)
            : base(container)
        {
        }
    }

    /*
    This class has been ported to TypeScript. See RoleGrid.ts
    Code below is only for reference purposes who wants to use Saltaralle 

    [ColumnsKey("Administration.Role"), IdProperty("RoleId"), NameProperty("RoleName")]
    [DialogType(typeof(RoleDialog)), LocalTextPrefix("Administration.Role"), Service("Administration/Role")]
    public class RoleGrid : EntityGrid<RoleRow>, IAsyncInit
    {
        public RoleGrid(jQueryObject container)
            : base(container)
        {
        }

        protected override List<string> GetDefaultSortBy()
        {
            return new List<string> { "RoleName" };
        }
    }
    */
}
 

File: Serene/Serene.Core/Modules/Administration/Role/RoleGrid.ts
namespace Serene.Administration {

    @Serenity.Decorators.registerClass()
    export class RoleGrid extends Serenity.EntityGrid<RoleRow, any> {
        protected getColumnsKey() { return "Administration.Role"; }
        protected getDialogType() { return RoleDialog; }
        protected getIdProperty() { return RoleRow.idProperty; }
        protected getLocalTextPrefix() { return RoleRow.localTextPrefix; }
        protected getService() { return RoleService.baseUrl; }

        constructor(container: JQuery) {
            super(container);
        }

        protected getDefaultSortBy() {
            return [RoleRow.Fields.RoleName];
        }
    }
}